go/secretsanta/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"log"
+	"math/rand"
+	"time"
+)
+
+var santaList = make([]Pair, 0)
+
+func main() {
+	names := [10]string{"Ally", "Gus",
+		"Roger", "Danil", "Isabel", "Pete",
+		"Rob", "Renee", "Shannon", "Greg"}
+	permutations := make([]Pair, 0)
+	shuffledPermutations := make([]Pair, 0)
+
+	// create all permutations
+	for i := 0; i < len(names); i++ {
```

Comment:
why create all permutations? why not check as you go? that will be a lot faster.